Phoenix/hello/lib/hello/catalog.ex

```diff
@@ -0,0 +1,219 @@
+defmodule Hello.Catalog do
+  @moduledoc """
+  The Catalog context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Hello.Repo
+  alias Hello.Catalog.Category
+  alias Hello.Catalog.Product
+
+  @doc """
+  Returns the list of products.
+
+  ## Examples
+
+      iex> list_products()
+      [%Product{}, ...]
+
+  """
+  def list_products do
+    Repo.all(Product)
+  end
+
+  @doc """
+  Gets a single product.
+
+  Raises `Ecto.NoResultsError` if the Product does not exist.
+
+  ## Examples
+
+      iex> get_product!(123)
+      %Product{}
+
+      iex> get_product!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_product!(id) do
+    Product |> Repo.get(id) |> Repo.preload(:categories)
```

Comment:
@samu975 what does a preload do?

Reply:
Preload, as its name says serves to preload, in this case, the categories. It allows us to reference them in templates, controllers, and views.

Reply:
🤔 seems it is not clear, please revisit the guides.